python/osrd_schemas/osrd_schemas/rolling_stock.py

```diff
@@ -169,6 +170,29 @@ class EtcsBrakeParams(BaseModel, extra="forbid"):
     t_be: float = Field(ge=0, description="T_be: safe brake build up time in s")
 
 
+class SimpleSignalingSystem(BaseModel, extra="forbid"):
+    signaling_system_type: Literal["BAL", "BAPR", "TVM300", "TVM430"] = Field(
+        alias="type"
+    )
+
+    model_config = ConfigDict(serialize_by_alias=True)
+
+
+class EtcsSignalingSystem(BaseModel, extra="forbid"):
+    signaling_system_type: Literal["ETCS_LEVEL2"] = Field(
+        alias="type", default="ETCS_LEVEL2"
+    )
+    brake_params: EtcsBrakeParams
+
+    model_config = ConfigDict(serialize_by_alias=True)
+
+
+SupportedSignalingSystem = Annotated[
+    Union[SimpleSignalingSystem, EtcsSignalingSystem],
+    Field(discriminator="type"),
```

Comment:
It breaks on client's side, this change is required:
```suggestion
    Field(discriminator="signaling_system_type"),
```

Reply:
done